wrapper.go

```diff
@@ -0,0 +1,135 @@
+package sfxlambda
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-lambda-go/lambdacontext"
+	"github.com/signalfx/golib/datapoint"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+)
+
+// HandlerFuncWrapper is the interface that wraps the lambda handler function.
+type HandlerFuncWrapper interface {
+	GetWrappedHandlerFunc() func(context.Context, interface{}) (interface{}, error)
+	SendDatapoint(context.Context, *datapoint.Datapoint)
+}
+
+type handlerFuncWrapper struct {
+	wrappedHandlerFunc func(context.Context, interface{}) (interface{}, error)
+	defaultDimensions  map[string]string
+	notColdStart       bool
+}
+
+// NewHandlerFuncWrapper is the HandlerFuncWrapper factory function.
+func NewHandlerFuncWrapper(handlerFunc interface{}) HandlerFuncWrapper {
+	hfw := handlerFuncWrapper{}
+	hfw.wrappedHandlerFunc = func(ctx context.Context, payload interface{}) (interface{}, error) {
+		var response interface{}
+		var err error
+		start := time.Now()
+		if hfw.defaultDimensions, err = getDefaultDimensions(ctx); err == nil {
+			hfw.sendInvocationsDatapoint()
+			hfw.sendColdStartsDatapoint()
+			var payloadBytes, responseBytes []byte
+			if payloadBytes, err = json.Marshal(payload); err == nil {
```

Comment:
I don't know much about Lambda but does it always require a JSON input payload?

Reply:
The input and output types have to be compatible with the "encoding/json" standard library.